### PR TITLE
fix: [MTL] Fix coverity issue

### PR DESCRIPTION
**Pull request by koktong-ong**
Fix for coverity scan issues

TEST: Able to boot with Windows and Ubuntu

### CODE REVIEW

**Review by jinjhuli — APPROVED** [review]

File: Platform/MeteorlakeBoardPkg/Library/FspsUpdUpdateLib/FspsUpdUpdateLib.c [hunk 1]
```diff
@@ -835,6 +802,36 @@ UpdateFspConfig (
...
+    FspsConfig->PcieRpUnsupportedRequestReport[Index]    = 0x1;
+    FspsConfig->PcieRpFatalErrorReport[Index]            = 0x1;
+    FspsConfig->PcieRpNoFatalErrorReport[Index]          = 0x1;
+    FspsConfig->PcieRpCorrectableErrorReport[Index]      = 0x1;
+    FspsConfig->PcieRpSnoopLatencyOverrideMode[Index]    = 0x2;
+    FspsConfig->PcieRpNonSnoopLatencyOverrideMode[Index] = 0x2;
+    FspsConfig->PcieRpPhysicalSlotNumber[Index]          = (UINT8)Index;
+  }
+  FspsConfig->PcieGen3EqPh3Preset5List[11]               = 0x9;
+  FspsConfig->PcieGen4EqPh3NoOfPresetOrCoeff[11]         = 0x5;
```
Comment (gdong1) [inline Platform/MeteorlakeBoardPkg/Library/FspsUpdUpdateLib/FspsUpdUpdateLib.c:833]: It looks this change is not related with Coverity issue as mentioned in the commit message. maybe split this PR into 2 PR?

Comment (koktong-ong) [reply Platform/MeteorlakeBoardPkg/Library/FspsUpdUpdateLib/FspsUpdUpdateLib.c:833]: This is the portion i moved from line 554. No new value was assigned to UPD.

**Review by ywho79 — APPROVED** [review]

File: Platform/MeteorlakeBoardPkg/Library/FspsUpdUpdateLib/FspsUpdUpdateLib.c (hunk 1 above)
Comment (koktong-ong) [reply Platform/MeteorlakeBoardPkg/Library/FspsUpdUpdateLib/FspsUpdUpdateLib.c]: patch updated

**Review by rprangar — APPROVED** [review]
LGTM
